### PR TITLE
[ADD] stock_pack_operation_quick_lot: New module

### PR DESCRIPTION
**Pull request by carlosdauden**
@Tecnativa

### CODE REVIEW

File: stock_pack_operation_quick_lot/README.rst
Comment (pedrobaeza) [inline stock_pack_operation_quick_lot/README.rst]: s/allows set/allows to set

File: stock_pack_operation_quick_lot/README.rst
Comment (pedrobaeza) [inline stock_pack_operation_quick_lot/README.rst]: s/transfer/transfers

File: stock_pack_operation_quick_lot/README.rst
Comment (pedrobaeza) [inline stock_pack_operation_quick_lot/README.rst]: s/create and/create it and click on

File: stock_pack_operation_quick_lot/README.rst
Comment (pedrobaeza) [inline stock_pack_operation_quick_lot/README.rst]: Add a new line saying: `#. Set optionally the end of life date in the proper column` and remove `[and date]`, and add a final line:
`#. When the picking will be transferred, the lot will be created with that name and date.`

File: stock_pack_operation_quick_lot/__openerp__.py
Comment (pedrobaeza) [inline stock_pack_operation_quick_lot/__openerp__.py]: Short this with `Set lot name and end date directly on picking operations`

File: stock_pack_operation_quick_lot/views/stock_view.xml
Comment (pedrobaeza) [inline stock_pack_operation_quick_lot/views/stock_view.xml]: Shouldn't this be visible only on incoming pickings?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: stock_pack_operation_quick_lot/views/stock_view.xml
Comment (carlosdauden) [reply stock_pack_operation_quick_lot/views/stock_view.xml]: Why?

Comment (pedrobaeza) [reply stock_pack_operation_quick_lot/views/stock_view.xml]: Because you are not creating a lot on outgoing. Lot name can be on outgoing, but definitively not end date.

File: stock_pack_operation_quick_lot/README.rst
Comment (rafaelbn) [inline stock_pack_operation_quick_lot/README.rst:25]: #. Activate "Track lots or serial numbers" in Inventory -> Configuration -> Setting ->  Lots and Serial Numbers 
#. Note: the product must have "Tracking | By Lots" in the inventory tab

File: stock_pack_operation_quick_lot/README.rst
Comment (rafaelbn) [inline stock_pack_operation_quick_lot/README.rst]: I'm not able to select any lot in this point.



Tested in http://3319581-410-b1204b.runbot2.odoo-community.org/web?debug=#id=20&view_type=form&model=stock.picking&menu_id=192&action=266

**Review by rafaelbn — COMMENTED** [review]
@carlosdauden thanks for this grear contribution, I've tested but i'm not able to select a lot. Check screenshot please

File: stock_pack_operation_quick_lot/README.rst
Comment (pedrobaeza) [reply stock_pack_operation_quick_lot/README.rst:25]: This is in configuration.

File: stock_pack_operation_quick_lot/README.rst
Comment (pedrobaeza) [reply stock_pack_operation_quick_lot/README.rst]: @rafaelbn of course, because it's for creating it on incoming receptions!!

Comment (rafaelbn) [reply stock_pack_operation_quick_lot/README.rst]: @pedrobaeza @carlosdauden shouldn't be hidden for outgoing pickings?

File: stock_pack_operation_quick_lot/__openerp__.py
Comment (rafaelbn) [reply stock_pack_operation_quick_lot/__openerp__.py]: [add] This module is for creating lot on incoming receptions

File: stock_pack_operation_quick_lot/README.rst
Comment (pedrobaeza) [reply stock_pack_operation_quick_lot/README.rst]: There's a technical glitch with that

**Comment by carlosdauden** [comment]
Changes done

File: stock_pack_operation_quick_lot/views/stock_view.xml
Comment (carlosdauden) [inline stock_pack_operation_quick_lot/views/stock_view.xml]: I don't know why this don't work...

File: stock_pack_operation_quick_lot/views/stock_view.xml
Comment (carlosdauden) [reply stock_pack_operation_quick_lot/views/stock_view.xml]: I try hide it

File: stock_pack_operation_quick_lot/README.rst
Comment (carlosdauden) [reply stock_pack_operation_quick_lot/README.rst:25]: Added in configuration

File: stock_pack_operation_quick_lot/views/stock_view.xml
Comment (pedrobaeza) [reply stock_pack_operation_quick_lot/views/stock_view.xml]: OK, let's continue with this hidden

**Comment by pedrobaeza** [comment]
Can you please complete tests for not dropping coverage?

**Comment by pedrobaeza** [comment]
@carlosdauden please finish this

**Review by pedrobaeza — APPROVED** [review]
Let's merge this one as is and improve on v11 as it has been a long time.

**Comment by rousseldenis** [comment]
@pedrobaeza Ok. I suppose you need this in v11. As model and workflow have changed, a backport of changes could be difficult. I would have let this open because you introduce a buggy module... See https://github.com/OCA/stock-logistics-workflow/pull/410#discussion_r173801550 I see also some computes that are not really correct (wrong depends)

At least, I would have written a known bugs section. You maybe have a too fast merge button reflex...

**Comment by pedrobaeza** [comment]
It's not a buggy module. It's just one feature that my colleague wanted to add that doesn't work. The module is working in our production customer since a long, and that's why I have merged it. Am I wrong, @carlosdauden?

And yes, we are migrating this already to v11, so expect soon the migration of the main feature, maybe in the same module, maybe integrated in an existing one.

**Comment by rousseldenis** [comment]
Ok, that's not clear in the conversation. @carlosdauden I'll make remarks here. Could you attend them in an new PR ? Thanks

File: stock_pack_operation_quick_lot/models/stock_pack_operation.py
Comment (rousseldenis) [inline stock_pack_operation_quick_lot/models/stock_pack_operation.py:29]: @carlosdauden Please depends on pack_lot_ids.lot_id.name.

**Comment by pedrobaeza** [comment]
OK, thanks for the follow-up

File: stock_pack_operation_quick_lot/models/stock_pack_operation.py
Comment (rousseldenis) [inline stock_pack_operation_quick_lot/models/stock_pack_operation.py:63]: Why not computing life_date based on pack_lot_ids? It could avoid multiple searches in the loop (if it depends on it).

File: stock_pack_operation_quick_lot/models/stock_pack_operation.py
Comment (rousseldenis) [inline stock_pack_operation_quick_lot/models/stock_pack_operation.py:69]: If you have multiple lots with multiple dates, it is a little bit cavalier to chose it like that. Because at the end, the life date you display can be random (there is no sort attribute).

I would have done a config on picking type to sort lot life date display asc or desc (or fixed to make it simplier)

File: stock_pack_operation_quick_lot/models/stock_pack_operation.py
Comment (rousseldenis) [inline stock_pack_operation_quick_lot/models/stock_pack_operation.py:95]: product_id.tracking

**Review by rousseldenis — COMMENTED** [review]
Code review.

My bigger doubt is on multiple lots on line management

**Comment by rousseldenis** [comment]
@carlosdauden Remarks done. I'll look after your changes PR and review it fast

**Comment by rousseldenis** [comment]
@pedrobaeza I see here there was no request of review. Even if the PR is old, please ask, eyes of some is preferrable than eyes of one :wink:

**Comment by pedrobaeza** [comment]
Are you talking about the label? Yes, sorry for that, although we are not using that flow. We review on demand, hehe.

**Comment by rousseldenis** [comment]
Maybe or before merging, if you ping repo maintainers they would come and see.

**Comment by pedrobaeza** [comment]
I will from now on in this repo knowing that you are there. There are a lot of repositories unattended and I eventually merge old PRs (from any contributor) that are green on CIs for not losing them.

**Pull request merged**
